src/main/java/bwapi/BWClientConfiguration.java

```diff
@@ -0,0 +1,77 @@
+     * as the bot has no prior access to BWAPI or game information.
+     *
+     * This flag indicates that taking arbitrarily long on frame zero is acceptable.
+     * Performance metrics omit the frame as an outlier.
+     * Asynchronous operation will block until the bot's event handlers are complete.
+     */
+    public boolean unlimitedFrameZero = true;
+
+    /**
+     * The maximum amount of time the bot is supposed to spend on a single frame.
+     * In asynchronous mode, JBWAPI will attempt to let the bot use up to this much time to process all frames before returning control to BWAPI.
+     * In synchronous mode, JBWAPI is not empowered to prevent the bot to exceed this amount, but will record overruns in performance metrics.
+     * Real-time human play typically uses the "fastest" game speed, which has 42.86ms (42,860ns) between frames.
+     */
+    public int maxFrameDurationMs = 40;
+
+    /**
+     * Runs the bot in asynchronous mode. Asynchronous mode helps attempt to ensure that the bot adheres to real-time performance constraints.
+     *
+     * Humans playing StarCraft (and some tournaments) expect bots to return commands within a certain period of time; ~42ms for humans ("fastesT" game speed),
+     * and some tournaments enforce frame-wise time limits (at time of writing, 55ms for COG and AIIDE; 85ms for SSCAIT).
+     *
+     * Asynchronous mode invokes bot event handlers in a separate thread, and if all event handlers haven't returned by a specified period of time, sends an
+     * returns control to StarCraft, allowing the game to proceed while the bot continues to step in the background. This increases the likelihood of meeting
+     * real-time performance requirements, while not fully guaranteeing it (subject to the whims of the JVM thread scheduler), at a cost of the bot possibly
+     * issuing commands later than intended, and a marginally larger memory footprint.
+     */
+    public boolean async = false;
+
+    /**
+     * The maximum number of frames to buffer while waiting on a bot.
+     * Each frame buffered adds about 33 megabytes to JBWAPI's memory footprint.
+     */
+    public int asyncFrameBufferSize = 10;
+
+    /**
+     * Toggles verbose logging, particularly of synchronization steps.
+     */
+    public boolean logVerbosely = false;
+
+    /**
+     * Checks that the configuration is in a valid state. Throws an IllegalArgumentException if it isn't.
+     */
+    public void validate() {
+        if (async && maxFrameDurationMs < 0) {
+            throw new IllegalArgumentException("maxFrameDurationMs needs to be a non-negative number (it's how long JBWAPI waits for a bot response before returning control to BWAPI).");
+        }
+        if (async && asyncFrameBufferSize < 1) {
+            throw new IllegalArgumentException("asyncFrameBufferSize needs to be a positive number (There needs to be at least one frame buffer).");
+        }
+    }
+
+    public void log(String value) {
```

Comment:
feels kind of weird to have this in the config but I get it, I would make the method package-private though